packages/react-components/react-link/src/components/Link/useLink.ts

```diff
@@ -15,8 +15,6 @@ export const useLink_unstable = (
 ): LinkState => {
   const backgroundAppearance = useBackgroundAppearance();
   const { appearance = 'default', disabled = false, disabledFocusable = false, inline = false } = props;
-  const as = props.as || (props.href ? 'a' : 'button');
```

Comment:
Why are we ignoring this condition now?

Reply:
Because this condition is already done internally by `getIntrinsicElementProps`